### PR TITLE
Changed to use photonvision

### PR DESCRIPTION
**Pull request by OliverW10**

### CODE REVIEW

File: components/vision.py [hunk 1]
```diff
@@ -38,135 +16,74 @@ class Vision:
...
                 return
-            # Come up with a position std dev from the fitness reported
-            # When the target is near the edge, the estimate of range is worse
-            pos_std_dev = self.max_std_dev * (1.0 - self.vision_data.fitness)
-            # TODO Can we be smarter and find different values for x and y based on robot orientation?
             self.chassis.estimator.addVisionMeasurement(
-                vision_pose,
-                self.vision_data.timestamp,
-                (pos_std_dev, pos_std_dev, 0.001),
+                Pose2d(vision_pose, self.chassis.get_rotation()),
```
Comment (auscompgeek) [inline components/vision.py:82]: Shouldn't this take into account the robot heading at the given time?

File: components/vision.py [hunk 2]
```diff
@@ -38,135 +16,74 @@ class Vision:
...
-        turret_angle = self.turret.get_angle_at(t)
-        chassis_heading = self.chassis.get_pose_at(t).rotation().radians()
-
-        # Ranges from vision are from centre of goal to the camera
-        # Add the offset to get range to the centre of the turret
-        range = self.vision_data.distance + self.CAMERA_OFFSET
-
-        vision_pose = pose_from_vision(
-            range, turret_angle + self.vision_data.angle, chassis_heading
+        vision_camera_pose = Translation2d(
```
Comment (auscompgeek) [inline components/vision.py:68]: 🤔 I thought the PhotonVision library has a method to return the pose

File: components/vision.py [hunk 3]
```diff
@@ -38,135 +16,74 @@ class Vision:
...
-    CAMERA_OFFSET = 0.35  # m from camera to centre of turret, measured from CAD
-    TURRET_OFFSET = 0.15  # m from robot centre to turret centre, measured from CAD
+    CAMERA_OFFSET = -0.1  # m from camera to centre of turret, measured from CAD
+    TURRET_OFFSET = -0.15  # m from robot centre to turret centre, measured from CAD
+
+    # camera angle from horizontal
+    CAMERA_PITCH = math.radians(20)
+    CAMERA_HEIGHT = 0.7
+    TARGET_HEIGHT = 2.66
+    GOAL_RAD = 0.61
```
Comment (auscompgeek) [inline components/vision.py:26]: I'm confused, what is this?

File: components/vision.py [hunk 4]
```diff
@@ -1,169 +1,105 @@
...
-            if self.gate_innovation and innovation > 3.0:
+            if self.gate_innovation and innovation > 5.0:
                 return
-            # Come up with a position std dev from the fitness reported
-            # When the target is near the edge, the estimate of range is worse
-            pos_std_dev = self.max_std_dev * (1.0 - self.vision_data.fitness)
-            # TODO Can we be smarter and find different values for x and y based on robot orientation?
+
+            std_dev = self.max_std_dev * min(
+                1, max(0, scale_value(self.distance, 5, 8, 1, 0.3))
```
Comment (auscompgeek) [inline components/vision.py:80]: Are the output standard deviations the wrong way around here? I would think the closer (i.e. more visible) the target is the more confident we'd be in the vision reading.

**Review by james-ward — APPROVED** [review]

**Pull request merged**